eZ/Publish/Core/Search/Elasticsearch/Content/Mapper/StandardMapper.php

```diff
@@ -348,6 +329,59 @@ protected function mapFields(Content $content, Type $contentType)
     }
 
     /**
+     * Get search fields for the given Content Field.
+     *
+     * @param \eZ\Publish\SPI\Persistence\Content\Type $contentType
+     * @param \eZ\Publish\SPI\Persistence\Content\Field $field
+     * @param \eZ\Publish\SPI\Persistence\Content\Type\FieldDefinition $fieldDefinition
+     * @param string $languageCode
+     *
+     * @return \eZ\Publish\SPI\Search\Field[]
+     */
+    protected function getSearchFields(
+        Type $contentType,
+        Content\Field $field,
+        Type\FieldDefinition $fieldDefinition,
+        $languageCode
+    ) {
+        $fieldType = $this->fieldRegistry->getType($field->type);
+        $searchFields = [];
+
+        $filterData = $fieldType->getFilterData($field, $fieldDefinition);
+        foreach ($filterData as $filterField) {
+            $searchFields[] = new Field(
+                $name = $this->fieldNameGenerator->getName(
+                    $filterField->name,
+                    $fieldDefinition->identifier,
+                    $contentType->identifier
+                ),
+                $filterField->value,
+                $filterField->type
+            );
+        }
+
+        if (!$fieldDefinition->isSearchable) {
+            return $searchFields;
+        }
+
+        $fullTextData = $fieldType->getFullTextData($field, $fieldDefinition);
+        foreach ($fullTextData as $fullTextValue) {
```

Comment:
`FullText` field is already handled by `MultipleStringField` mapper, so it would be OK to construct a single field passing `$fullTextData` (an array of strings) to it derectly.